plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleConnectorTest.java

```diff
@@ -83,4 +121,38 @@ protected SqlExecutor onRemoteDatabase()
     {
         return oracleServer::execute;
     }
+
+    @Test
+    public void testPushdownJoinWithLongNameSucceedsWhenConfiguredCorrectly()
+    {
+        testPushdownJoinWithLongName(30);
```

Comment:
this is prone to breakage. e.g. in future when we upgrade to Oracle with 128 char limits how would one find what value to adjust?

You can look at `BaseConnectorTest#testCreateSchemaWithLongName` for some ideas. The other benefit being that then the test can be run against all connectors and we can catch such silent failures in other connectors too.

(This can be left as a follow-up, if you do decide to skip this comment then please create a GitHub issue + add a TODO comment linking to it)

Reply:
Actually this test is fine, as it formats the alias over 30 characters which (we believe) is below the limit for any database, no matter if it's Oracle 11.x or Postgres.

I think what you were referring to is the second test that uses the constant 31, and that specifically tries to break it for Oracle 11.x. If we upgraded to 12.x and the limit by doing that changes to 128, the the test will fail, as we would not get the error message from the SQL engine. So the engineer doing the Oracle upgrade will notice that he needs to fix the constant in the test.

If I got you wrong and you were actually talking about this specific test, then it is the case that if we upgrade to 12.x or extend to any other database, the test will still be passing and that is OK. We are fine with alias being squashed to 30 characters even if the actual DB engine allows the alias to have 128 characters. As a reminder we did agree on one of the daily sessions what was the acceptance criteria and 'out of scope' for this task, which I copy here for reference from our internal ticket:

Acceptance Criteria
- Solution fixes the bug for Oracle pushdown JOIN

Out of Scope
- Integration test for other databases than Oracle
- If Alias field restriction has same bug on aggregation, it won’t be fixed
- If other databases have restriction on the alias field that result in same bug, this wont' be fixed

Reply:
in addition to magic constant the main problem is that you have two tests which have two constants which can go out of sync.

It also makes it harder to make the test generic because we need to do it in future anyway. As an example if you strictly consider only Oracle to be in scope an even smaller solution would be to add a different implementation of `DefaultQueryBuilder` for Oracle instead of trying to add a generic formatter class. But that would make making this solution generic harder since everything will need to be reworked anyway.

So since we already know the long term goal the short term solutions should iteratively drive us towards those long term goals.

Reply:
@hashhar, thanks for the explanations. However, I do find it hard to skin out exactly what your asking for. There's to many 'ifs' for me here. Can you lay out in plain the changes/steps you want me to apply to this piece of code as if you were doing it? You can make the decisions instead of describing conditionals.

Reply:
@hubertsk, @hashhar. I tried to guess what your ask is. I have used a constant and a constant + 1, however I have a feeling that we are not totally in sync about this specific request. Do you think it's possible that we discuss that on-line?

Reply:
look at https://github.com/trinodb/trino/blob/398b64671cc2ecc1c23556d80eb20f29c4eca5eb/testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java#L3326-L3335 and try to model similarly if for nothing then at-least consitency's sake (makes future readers of the code's job easier since knowledge is transferable and things are familiar).

(There are benefits other than consistency as I've already said since as a follow up we need to do similar fix in other connectors and this change makes it possible to reuse the test instead of write it again for all of the more than 10 JDBC based connectors we have.)

Reply:
This code is now simplified after removing the negative test. If there's something that makes it unreadable, please propose the exact changes you want me to do and I will copy-paste them to the PR.